pkg/util/camel/catalog.go

```diff
@@ -91,13 +93,111 @@ func (c Catalog) VisitArtifacts(visitor func(string, Artifact) bool) {
 }
 
 // VisitSchemes --
-func (c Catalog) VisitSchemes(visitor func(string, Scheme) bool) {
+func (c RuntimeCatalog) VisitSchemes(visitor func(string, Scheme) bool) {
 	for id, scheme := range c.schemesByID {
 		if !visitor(id, scheme) {
 			break
 		}
 	}
 }
 
+// ******************************
+//
+//
+//
+// ******************************
+
+var defaultCatalog RuntimeCatalog
+var catalogs map[string]RuntimeCatalog
+var catalogsLock sync.Mutex
+
+func init() {
+	c, err := loadCatalog("camel-catalog.yaml")
+	if err != nil {
+		panic(err)
+	}
+
+	defaultCatalog = *c
+
+	catalogs = make(map[string]RuntimeCatalog)
+}
+
 // Runtime --
-var Runtime Catalog
+func loadCatalog(resourceName string) (*RuntimeCatalog, error) {
+	var catalog Catalog
+
+	data, ok := deploy.Resources[resourceName]
+	if !ok {
+		return nil, nil
+	}
+
+	if err := yaml.Unmarshal([]byte(data), &catalog); err != nil {
+		return nil, err
+	}
+
+	rc := RuntimeCatalog{}
+	rc.Catalog = catalog
+	rc.artifactByScheme = make(map[string]string)
+	rc.schemesByID = make(map[string]Scheme)
+
+	for id, artifact := range catalog.Artifacts {
+		for _, scheme := range artifact.Schemes {
+			scheme := scheme
+			rc.artifactByScheme[scheme.ID] = id
+			rc.schemesByID[scheme.ID] = scheme
+		}
+	}
+
+	return &rc, nil
+}
+
+// Runtime --
+func Runtime(camelVersion string) *RuntimeCatalog {
```

Comment:
I wonder whether the logic should be reversed. From an end-user standpoint, being able to specify a version in the platform spec following semver, and best matching the corresponding catalog, instead of matching x.y.z version with catalog x.y or x. I would find it more logical, but just a thought :)

Reply:
yes it make sense but I think it depends to the availability of different catalogs that we do not build by default now, do you mind rising an issue ?